src/robot/running/arguments/argumentspec.py

```diff
@@ -137,14 +164,15 @@ def required(self):
         return False
 
     @property
-    def type_repr(self):
-        if self.type is self.NOTSET:
-            return None
-        if isclass(self.type):
-            if issubclass(self.type, Enum):
-                return self._format_enum(self.type)
-            return self.type.__name__
-        return unicode(self.type)
+    def types_reprs(self):
+        return [self._type_repr(t) for t in self.types]
+
+    def _type_repr(self, typ):
+        if typ == type(None):
+            return 'None'
+        if isclass(typ):
+            return typ.__name__
+        return re.sub(r'^typing\.(.+)', lambda m: m.group(1), unic(typ))
```

Comment:
Dropping the `typing.` prefix isn't related to rest of this PR but it's so simple that having it here isn't a big problem. It would need some tests, but I guess it's fine to do that as part of #3724 as well.

I'd in general rather use `if type_repr.startswith('typing.'): return type_repr[7:]` than `re.sub`, but in this particular case that would be somewhat more code. Please replace the lambda with `r'\1'` anyway.